packages/backend/src/peripherals/database/EventRepository.ts

```diff
@@ -0,0 +1,80 @@
+import { Logger, ProjectId, UnixTime } from '@l2beat/common'
+import { EventRow } from 'knex/types/tables'
+
+import { BaseRepository } from './shared/BaseRepository'
+import { Database } from './shared/Database'
+
+export interface EventRecord {
+  timestamp: UnixTime
+  name: string
+  projectId: ProjectId
+  count: number
+  timeSpan: 'hourly' | 'sixHourly' | 'daily'
+}
+
+export class EventRepository extends BaseRepository {
+  constructor(database: Database, logger: Logger) {
+    super(database, logger)
+
+    /* eslint-disable @typescript-eslint/unbound-method */
+
+    this.addMany = this.wrapAddMany(this.addMany)
+    this.getAll = this.wrapGet(this.getAll)
+    this.deleteAll = this.wrapDelete(this.deleteAll)
+    this.getByProjectAndName = this.wrapGet(this.getByProjectAndName)
+
+    /* eslint-enable @typescript-eslint/unbound-method */
+  }
+
+  async getByProjectAndName(
+    projectId: ProjectId,
+    name: string,
```

Comment:
We probably would like to allow specifying multiple names at once

Reply:
Do you mean event names, or both event names and projectId?

Reply:
event names